### ISSUE TITLE
Building with Gradle gives a warning about CDI dependencies

### ISSUE THREAD

**Issue by geoand** [quarkusio/quarkus#4960]
Now that we have moved to Jakarta dependencies and because gradle leaks CDI dependencies into our classpath, we get the following warning when building a gradle project:

```
building native image
These dependencies are not recommended:
        - javax.enterprise:cdi-api should be replaced by jakarta.enterprise:jakarta.enterprise.cdi-api
        - javax.inject:javax.inject should be replaced by jakarta.inject:jakarta.inject-api
You might end up with two different versions of the same classes or with an artifact you shouldn't have in your classpath.
```

cc @gsmet @stuartwdouglas @emmanuelbernard

**Comment by stuartwdouglas** [comment]
We need class loader isolation, we can't have gradle leaking all its deps into our class path.

**Comment by geoand** [comment]
This might be due to how gradle is resolving dependencies. I am thinking this because I am seeing this part of the dependency try in a gradle project:

```
|    |    +--- io.quarkus:quarkus-core:999-SNAPSHOT
|    |    |    +--- jakarta.annotation:jakarta.annotation-api:1.3.5
|    |    |    +--- jakarta.enterprise:jakarta.enterprise.cdi-api:2.0.2
|    |    |    |    +--- jakarta.el:jakarta.el-api:3.0.3
|    |    |    |    +--- jakarta.interceptor:jakarta.interceptor-api:1.2.5
|    |    |    |    |    +--- jakarta.annotation:jakarta.annotation-api:1.3.5
|    |    |    |    |    \--- jakarta.ejb:jakarta.ejb-api:3.2.6
|    |    |    |    |         \--- jakarta.transaction:jakarta.transaction-api:1.3.2 -> 1.3.3
|    |    |    |    \--- jakarta.inject:jakarta.inject-api:1.0
|    |    |    +--- jakarta.inject:jakarta.inject-api:1.0
|    |    |    +--- io.smallrye:smallrye-config:1.3.9
|    |    |    |    +--- org.eclipse.microprofile.config:microprofile-config-api:1.3
|    |    |    |    +--- javax.enterprise:cdi-api:2.0.SP1
|    |    |    |    |    +--- javax.el:javax.el-api:3.0.0
|    |    |    |    |    +--- javax.interceptor:javax.interceptor-api:1.2
|    |    |    |    |    \--- javax.inject:javax.inject:1
|    |    |    |    \--- org.jboss.logging:jboss-logging:3.4.0.Final -> 3.3.2.Final
```

which seems to indicate that gradle is honoring the exclusions we have in quarkus-core:

```
        <dependency>
            <groupId>io.smallrye</groupId>
            <artifactId>smallrye-config</artifactId>
            <exclusions>
                <exclusion>
                    <groupId>javax.enterprise</groupId>
                    <artifactId>cdi-api</artifactId>
                </exclusion>
                <exclusion>
                    <groupId>javax.annotation</groupId>
                    <artifactId>javax.annotation-api</artifactId>
                </exclusion>
            </exclusions>
        </dependency>
```

**Comment by killboard** [comment]
Any update on this? Still warning when build native images
`javax.enterprise:cdi-api should be replaced by jakarta.enterprise:jakarta.enterprise.cdi-api`

**Comment by geoand** [comment]
This was fixed in `1.3.x`
